src/native/runtime-base/monodroid-dl.hh

```diff
@@ -112,7 +159,7 @@ namespace xamarin::android::internal
 
 			hash_t name_hash = xxhash::hash (name, strlen (name));
 			log_debug (LOG_ASSEMBLY, "monodroid_dlopen: hash for name '%s' is 0x%zx", name, name_hash);
-			DSOCacheEntry *dso = find_dso_cache_entry (name_hash);
+			DSOCacheEntry *dso = use_aot_cache ? find_any_dso_cache_entry (name_hash) : find_only_dso_cache_entry (name_hash);
```

Comment:
If `use_aot_cache` is true, shouldn't this invoke `find_only_aot_cache_entry()`?

Reply:
As per [my later comment](https://github.com/dotnet/android/pull/9117#discussion_r1731507271), this line is probably fine, it's just that the name `use_aot_cache` was misleading/confusing me.

Assuming my later comment is correct, that `use_aot_cache` means "prefer the AOT cache" instead of an either/or interpretation, then I think *removing* `find_any_dso_cache_entry()` entirely might be useful, then updating this callsite to:

```cpp
DSOCacheEntry *dso = nullptr;
if (prefer_aot_cache) {
    dso = find_only_aot_cache_entry (name_hash);
}
if (dso == nullptr) {
    dso = find_only_dso_cache_entry (name_hash);
}
```

which would help emphasize this is a "prefer AOT/look there first" vs. an "either/or" scenario.

Reply:
> If `use_aot_cache` is true, shouldn't this invoke `find_only_aot_cache_entry()`?

No, because the AOT context is less, hmm, defined.  The DSO cache entry context is 100% clear - it is always used from within the p/invoke resolution code, so we know the caller (MonoVM) won't be interested in anything else but libraries that contain p/invokable code.  When `monodroid_dlopen` is called, OTOH, the runtime might want either the AOT shared libraries or standard shared libraries and we don't know which.

Reply:
@grendello [wrote](https://github.com/dotnet/android/pull/9117#discussion_r1731638825):

> the runtime might want either the AOT shared libraries or standard shared libraries and we don't know which.

which is part of my confusion: yes, MonoVM wants either/or, but from *our* perspective it means we need to check *both*, in a particular order.

Maybe it's just me -- it's probably just me -- but "either/or" in combination with "use_aot_cache" implies *to me* that *either* the AOT cache should be used, *or* the DSO cache, *but **not** both*.

Meanwhile, we *do* want "both"!  In a particular preferential order!  Hence my suggestion to rename "use_aot_cache" to "prefer_aot_cache".

Reply:
It's `prefer_aot_cache` now.